test/integration/etcd_storage_path_test.go

```diff
@@ -0,0 +1,1010 @@
+package integration
+
+import (
+	"encoding/json"
+	"fmt"
+	"mime"
+	"net/http"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kubeerr "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/diff"
+	"k8s.io/kubernetes/pkg/util/flowcontrol"
+
+	"github.com/openshift/origin/pkg/api/latest"
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+
+	// install all APIs
+	_ "github.com/openshift/origin/pkg/api/install"
+
+	etcd "github.com/coreos/etcd/client"
+	"golang.org/x/net/context"
+
+	authorizationapiv1 "github.com/openshift/origin/pkg/authorization/api/v1"
+	sdnapi "github.com/openshift/origin/pkg/sdn/api"
+)
+
+// Etcd data for all persisted objects.  Be very careful when setting ephemeral to true as that removes the safety we gain from this test.
+var etcdStorageData = map[unversioned.GroupVersionResource]struct {
+	ephemeral        bool           // Set to true to skip testing the object
```

Comment:
no need to put ephemeral as a field here. have a separate map of ephemeral types and check GVRs missing from etcdStorageData against that list

that keeps this map clean and avoids the possibility of a copy/paste from this list copying an ephemeral case